### PR TITLE
Logged In User's Name

### PR DESCRIPTION
**Pull request by taylor-noel**

### CODE REVIEW

**Comment by Niyousha99** [comment]
Does the username state in sparkpage.tsx get updated accordingly when the user signs in/out? If there is no user signed in, what is the value of username (empty string, null, etc.)?

File: src/components/TechSpark/Login.tsx [hunk 1]
```diff
@@ -91,17 +93,9 @@ function Login(props) {
...
-                    <p className = "alertMessage">You have entered an invalid username</p>
-                    </div>  : <div />}
-                {invalidPassword ?
-                    <div className = "alert">
-                        <p className = "alertMessage">You have entered an invalid password</p>
-                    </div> : <div />}
                 {invalidLogin ?
                     <div className = "alert">
-                        <p className = "alertMessage">You have entered an invalid username or password</p>
+                        <p className = "alertMessage">{message}</p>
```
Comment (Jhappy77) [inline src/components/TechSpark/Login.tsx:98]: I like the message being dynamic! :)

File: src/components/TechSpark/Login.tsx [hunk 2]
```diff
@@ -1,59 +1,61 @@
...
     const [tracker, setTracker] = React.useState(false)
-    const [invalidUsername, setInvalidUsername] = React.useState(false)
-    const [invalidPassword, setInvalidPassword] = React.useState(false)
+    // const [invalidUsername, setInvalidUsername] = React.useState(false)
+    // const [invalidPassword, setInvalidPassword] = React.useState(false)
     const [invalidLogin, setInvalidLogin] = React.useState(false)
+    const [message, setMessage] = React.useState("")
     const firstUpdate = useRef(true);
     useLayoutEffect(() => {
+        let username = document.getElementById('loginUsername')
```
Comment (Jhappy77) [inline src/components/TechSpark/Login.tsx:16]: I don't like keeping track of the username/password this way. let's useState for the username & password. 

ex. something like 

const [name, setName] = useState("")
<input value={name} onChange={(e)=>{e.preventDefault(); setName(e.target.value)}}... />

you'll need to strong type e (the event). I think it's InputEvent or something like that. If you search it up you should be able to find it.

File: src/pages/SparkPage.tsx [hunk 3]
```diff
@@ -143,9 +144,13 @@ function Dashboard() {
...
     }
 
+    function handleLogin(name: string){
+        setUserName(name);
+    }
+
     return (
         <div className="sparkBackground" id="sparkPageTop">
-            {displayPost && <Popup windowType={windowType} hidePost={hidePost} />}
+            {displayPost && <Popup windowType={windowType} hidePost={hidePost} setName = {handleLogin}/>}
```
Comment (Jhappy77) [inline src/pages/SparkPage.tsx:153]: hmm why is setName equalling handleLogin? It's a good idea for names to match each other (and describe exactly what they do). This is a little confusing

**Review by Jhappy77 — CHANGES_REQUESTED** [review]
I have 2 suggestions for before you merge this, see my comments :)

File: src/components/TechSpark/Login.tsx (hunk 2 above)
Comment (Niyousha99) [reply src/components/TechSpark/Login.tsx:16]: You could also use useReducer to handle related states like username and password

**Pull request closed without merging**